packages/vite/src/node/optimizer/scan.ts

```diff
@@ -38,7 +38,7 @@ const htmlTypesRE = /\.(html|vue|svelte)$/
 // since even missed imports can be caught at runtime, and false positives will
 // simply be ignored.
 const importsRE =
-  /\bimport(?!\s+type)(?:[\w*{}\n\r\t, ]+from\s*)?\s*("[^"]+"|'[^']+')/gm
+  /\bimport(?!\s+type)(?:[\w*{}\n\r\t, ]+from\s*)?\s*("[^"]+"|'[^']+')$/gm
```

Comment:
This won't work in case there is a comment, a trailing `;` or spaces after the path.
Maybe we can use this idea though to filter out these cases, with something like:
```suggestion
  /(?:^|;|\*\/)\s*import(?!\s+type)(?:[\w*{}\n\r\t, ]+from\s*)?\s*("[^"]+"|'[^']+')\s*(?:$|;|\/\/|\/\*)/gm
```
This should allow for comments before or after the import, but failing in the reported case. It would still fail but it should do so in fewer cases. Playground here https://regexr.com/6125s
It would be good to have more test cases, better as unit tests of this regex in particular.

Reply:
make senses, updated